src/main/java/it/rate/webapp/controllers/InterestAdminController.java

```diff
@@ -30,11 +28,7 @@ public class InterestAdminController {
   @GetMapping("/edit")
   @PreAuthorize("hasAnyAuthority(@permissionService.manageCommunity(#interestId))")
   public String editInterestPage(@PathVariable Long interestId, Model model, Principal principal) {
-    Optional<Interest> interest = interestService.findInterestById(interestId);
-    if (interest.isEmpty()) {
-      throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Interest not found");
-    }
-    model.addAttribute("interest", interest.get());
+    model.addAttribute("interest", interestService.getById(interestId));

```

Comment:
is it 100% certain that the interest will exist? If not, the previous version with optional was better

Reply:
It should be 100%, as the manageCommunity method inside Preauthorize annotation gets executed first, and checks if Interest with given id exists and if not, throws an exception.